### PR TITLE
feat(userSelf): add default group name in user self api

### PR DESCRIPTION
**Pull request by Aman-Codes**
## Description

Added default group name information in `users/self` api

### Changes

- Added default group name information in `users/self` api
- Updated schema and example value in `openapi.yaml` file
- Updated API version from v1 to v2

### Postman response
image
### Updated example value in `openapi.yaml`
image
### Updated schema in `openapi.yaml`
image

## How to test

Make get request to `http://localhost/repo/api/v2/users/self` api

### CODE REVIEW

**Comment by Aman-Codes** [comment]
Fixed the codesniffer warnings. Ready for review now

File: src/www/ui/api/documentation/openapi.yaml [hunk 1]
```diff
@@ -681,13 +681,22 @@ paths:
...
-                $ref: '#/components/schemas/User'
+                allOf:
+                  - type: object
+                    properties:
+                      defaultGroup:
+                        type: string
+                        description: Default group of the user
+                        example:
+                          "fossy"
+                  - $ref: '#/components/schemas/User'
```
Comment (GMishx) [inline src/www/ui/api/documentation/openapi.yaml]: Please move `$ref` to top so `defaultGroup` comes at the bottom in documentation (as in the response).

File: src/www/ui/api/documentation/openapi.yaml [hunk 2]
```diff
@@ -681,13 +681,22 @@ paths:
...
         '200':
           description: OK
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/User'
+                allOf:
+                  - type: object
+                    properties:
+                      defaultGroup:
```
Comment (GMishx) [inline src/www/ui/api/documentation/openapi.yaml]: Value returned from code is `default_group`. Please update the doc accordingly.

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/www/ui/api/documentation/openapi.yaml (hunk 2 above)
Comment (Aman-Codes) [reply src/www/ui/api/documentation/openapi.yaml]: Updated it

File: src/www/ui/api/documentation/openapi.yaml (hunk 1 above)
Comment (Aman-Codes) [reply src/www/ui/api/documentation/openapi.yaml]: Updated it

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**